### ISSUE TITLE
gcc-14: incompatible-pointer-types

### ISSUE THREAD

**Issue by uwabami** [gfd-dennou-club/ruby-mpi#11]
```
compiling mpi.c
In file included from /usr/include/ruby-3.1.0/ruby/ruby.h:26,
                 from /usr/include/ruby-3.1.0/ruby.h:38,
                 from mpi.c:4:
mpi.c: In function 'Init_mpi':
/usr/include/ruby-3.1.0/ruby/internal/anyargs.h:337:142: error: passing argument 3 of 'rb_define_module_function_m1' from incompatible pointer type [-Wincompatible-pointer-types]
  337 | #define rb_define_module_function(mod, mid, func, arity)    RBIMPL_ANYARGS_DISPATCH_rb_define_module_function((arity), (func))((mod), (mid), (func), (arity))
      |                                                                                                                                              ^~~~~~
      |                                                                                                                                              |
      |                                                                                                                                              VALUE (*)(VALUE) {aka long unsigned int (*)(long unsigned int)}
mpi.c:697:3: note: in expansion of macro 'rb_define_module_function'
  697 |   rb_define_module_function(mMPI, "Finalize", rb_m_finalize, -1);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/ruby-3.1.0/ruby/internal/anyargs.h:273:21: note: expected 'VALUE (*)(int,  union <anonymous>,  VALUE)' {aka 'long unsigned int (*)(int,  union <anonymous>,  long unsigned int)'} but argument is of type 'VALUE (*)(VALUE)' {aka 'long unsigned int (*)(long unsigned int)'}
  273 | RBIMPL_ANYARGS_DECL(rb_define_module_function, VALUE, const char *)
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/ruby-3.1.0/ruby/internal/anyargs.h:253:41: note: in definition of macro 'RBIMPL_ANYARGS_DECL'
  253 | RBIMPL_ANYARGS_ATTRSET(sym) static void sym ## _m1(__VA_ARGS__, VALUE(*)(int, union { VALUE *x; const VALUE *y; } __attribute__((__transparent_union__)), VALUE), int); \
      |                                         ^~~
make: *** [Makefile:246: mpi.o] Error 1
```

**Comment by seiya** [comment]
fixed in b476162

**Comment by uwabami** [comment]
Thx!
